app/models/foreman_tasks/task/dynflow_task.rb

```diff
@@ -97,16 +103,38 @@ def cli_example
 
     def main_action
       return @main_action if defined?(@main_action)
-      @main_action = execution_plan && execution_plan.root_plan_step.try(:action, execution_plan)
+      if active_job?
+        action = execution_plan_action
+        active_job_action(action.input['job_class'], action.input['job_arguments'])
```

Comment:
If the action is `:scheduled` or `:planning`, the input is still unset, leading to 
```
2018-01-22T13:11:17  [dynflow] [W] Error on on_execution_plan_save event
 | NoMethodError: undefined method `constantize' for nil:NilClass
 | /home/aruzicka/git/foreman-tasks/app/models/foreman_tasks/task/dynflow_task.rb:118:in `active_job_action'
 | /home/aruzicka/git/foreman-tasks/app/models/foreman_tasks/task/dynflow_task.rb:108:in `main_action'
```

Reply:
How did you get this error, for example by scheduling a REX job ?

Reply:
I think I scheduled an active job to be executed in the future and went to look into the tasks index before it was started